modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/StandaloneASDiscovery.java

```diff
@@ -82,17 +82,28 @@ protected String getLogFileName() {
     @Override
     protected String buildDefaultResourceName(HostPort hostPort, HostPort managementHostPort,
         JBossProductType productType, String serverName) {
+        return super.buildDefaultResourceName(hostPort, managementHostPort, productType, serverName);
+    }
+
+    @Override
+    String buildDefaultResourceName(HostPort hostPort, HostPort managementHostPort,
+        JBossProduct product, String serverName) {
         if (serverName != null && !serverName.trim().isEmpty()) {
-            return String.format("%s (%s %s:%d)", productType.SHORT_NAME, serverName, managementHostPort.host,
+            return String.format("%s (%s %s:%d)", product.SHORT_NAME, serverName, managementHostPort.host,
                 managementHostPort.port);
         }
 
-        return String.format("%s (%s:%d)", productType.SHORT_NAME, managementHostPort.host, managementHostPort.port);
+        return String.format("%s (%s:%d)", product.SHORT_NAME, managementHostPort.host, managementHostPort.port);
     }
 
     @Override
     protected String buildDefaultResourceDescription(HostPort hostPort, JBossProductType productType) {
-        return String.format("Standalone %s server", productType.FULL_NAME);
+        return super.buildDefaultResourceDescription(hostPort, productType);
+    }
+
+    @Override
+    public String buildDefaultResourceDescription(HostPort hostPort, JBossProduct product) {
```

Comment:
Do not augment visibility (keep it package protected like in the parent class), to avoid future problems with API compatibility.